clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+    return false;
+  }
+
+  debugLog("No location-specific rules matched, assume banner must be shown. isBannerEnabledForLocation = true");
+  return true;
+}
+
+/**
+ * Builds a button DOM element with the given id, class name, and text label
+ */
+const buildButton = (id: string, className: string, label?: string, onClick?: (event: MouseEvent) => void): HTMLButtonElement => {
+  const button = document.createElement("button");
+  button.id = id;
+  button.innerHTML = label || "";
+  button.classList.add("fides-consent-banner-button");
+  button.classList.add(className);
+  button.addEventListener("click", (event) => {
+    debugLog(`Fides consent banner button clicked with id='${id}'`);
+    if (onClick !== undefined) {
+      onClick(event);
+    }
+  });
+  return button;
+};
+
+/**
+ * Show the banner
+ */
+const showBanner = (banner: HTMLDivElement) => {
+  banner.classList.remove("fides-consent-banner-hidden");
+};
+
+/**
+ * Hide the banner (probably because the user selected a consent option)
+ */
+const hideBanner = (banner: HTMLDivElement) => {
+  banner.classList.add("fides-consent-banner-hidden");
+};
+
+/**
+ * Builds the DOM elements for the consent banner (container, buttons, etc.) and
+ * return a single div that can be added to the body.
+ */
+const buildBanner = (defaults: CookieKeyConsent): HTMLDivElement => {
+  const options: ConsentBannerOptions = getBannerOptions();
+
+  // Create the overall banner container
+  const banner = document.createElement("div");
+  banner.id = "fides-consent-banner";
+  banner.classList.add("fides-consent-banner");
+  // TODO: support option to specify top/bottom
+  banner.classList.add("fides-consent-banner-bottom");
+  banner.classList.add("fides-consent-banner-hidden");
+
+  // Add the banner description
+  const bannerDescription = document.createElement("div");
+  bannerDescription.id = "fides-consent-banner-description";
+  bannerDescription.classList.add("fides-consent-banner-description");
+  bannerDescription.innerText = options.labels?.bannerDescription || "";
+  banner.appendChild(bannerDescription);
+
+  // Create the button container
+  const buttonContainer = document.createElement("div");
+  buttonContainer.id = "fides-consent-banner-buttons";
+  buttonContainer.classList.add("fides-consent-banner-buttons");
+
+  // Create the banner buttons
+  const tertiaryButton = buildButton(
+    "fides-consent-banner-button-tertiary",
+    "fides-consent-banner-button-tertiary",
+    options.labels?.tertiaryButton,
+    navigateToPrivacyCenter,
+  );
+  buttonContainer.appendChild(tertiaryButton);
+  const secondaryButton = buildButton(
+    "fides-consent-banner-button-secondary",
+    "fides-consent-banner-button-secondary",
+    options.labels?.secondaryButton,
+    () => {
+      setConsentCookieRejectAll(defaults);
+      hideBanner(banner);
+      // TODO: save to Fides consent request API
```

Comment:
Really not sure of the right way to do this... but we can't forget the importance of supporting this to stay on-parity with the Privacy Center.